dojo/tools/semgrep/parser.py

```diff
@@ -0,0 +1,51 @@
+import json
+from dojo.models import Finding
+from dojo.tools.semgrep.models import SemgrepJSONResult
+
+
+class SemgrepJSONParser(object):
+
+    def __init__(self, filehandle, test):
+        tree = self.parse_json(filehandle)
+
+        self.items = []
+        if tree:
+            results = tree.get('results')
+
+            if not results:
+                return
+
+            for item in results:
+                title = item['check_id']
+                f = SemgrepJSONResult(item['extra'], item['path'], item['start'], item['end'])
+
+                findingItem = Finding(
+                    title=f.title,
+                    severity=f.severity,
+                    numerical_severity=Finding.get_numerical_severity(f.severity),
+                    description=f.message,
+                    mitigation='N/A',
+                    file_path=item['path'],
+                    cwe=f.cwe,
+                    line=f.start,
+                    active=False,
+                    verified=False,
+                    url='N/A',
```

Comment:
No need to specify `N/A` if nothing to input. 
However, could `self.source_rule_url` be used to fill in that field?

Reply:
not sure, because in my understanding of DojoFindings, url is related to a test url ? did I miss something ?

Reply:
Last update to what is was is "pointing to an external source to explain the finding". Otherwise, please remove altogether.